modules/ingress-setting-a-custom-default-certificate.adoc

```diff
@@ -0,0 +1,72 @@
+// Module included in the following assemblies:
+//
+// * ingress/ingress-operator.adoc
+
+[id='ingress-setting-a-custom-default-certificate-{context}']
+= Setting a Custom Default Certificate
+
+You can configure an ingress controller to use a custom certificate by creating a `Secret` resource and editing the `IngressController` custom resource (CR).
+
+.Prerequisites
+
+* You must have a certificate/key pair in PEM-encoded files, where the certificate is signed by a trusted certificate authority and valid for the ingress domain.
+
+* You must have an `IngressController` CR.  You may use the default one:
++
+----
+$ oc -n openshift-ingress-operator get IngressControllers
+NAME      AGE
+default   10m
+----
+
+.Procedure
+
+The following assumes that the custom certificate/key pair are in the `tls.crt` and `tls.key` files in the current working directory.  Substitute the actual path names for `tls.crt` and `tls.key`.  You also may substitute another name for the `custom-certs-default` when creating the `Secret` resource and referencing it in the `IngressController` CR.
+
+First, create a `Secret` resource with the custom certificate in the `openshift-ingress` namespace:
+
+. Create a `Secret` resource using the `tls.crt` and `tls.key` files.
++
+----
+$ oc -n openshift-ingress create secret tls custom-certs-default --cert=tls.crt --key=tls.key
+----
+
+Then modify the `IngressController` CR:
+
+. Open the `IngressController` CR for editing.
++
+----
+$ oc -n openshift-ingress-operator edit ingresscontrollers/default
+----
++
+For example:
++
+----
+oc -n openshift-ingress-operator edit ingresscontrollers/default
+
+apiVersion: ingresscontroller.operator.openshift.io/v1
+kind: IngressController
+metadata:
+  creationTimestamp: 2019-03-04T00:18:48Z
+  generation: 1
+  name: default
+  namespace: openshift-ingress-operator
+  resourceVersion: "15872"
+  selfLink: /apis/operator.operator.openshift.io/v1/namespaces/openshift-ingress-operator/ingresscontrollers/default
+  uid: e8f5dc3c-5300-43d5-a0c7-48ebf228634f
+namespace: openshift-ingress-operator
+----
+
+. Change the `defaultCertificate` parameter:
++
+.Sample Ingress Controller
+[source,yaml]
+----
+spec:
+  defaultCertificate:
+    name: "custom-certs-default" <1>
+----
+
+<1> Name of the `Secret` resource in the `openshift-ingress` namespace with the TLS secret for the ingress controller.
+
+As soon as the `IngressController` CR has been modified, the ingress operator will update the ingress controller's deployment to use the custom certificate.  Note that this action will cause the ingress controller to restart.
```

Comment:
>Note that this action will cause the ingress controller to restart.

Well, sort of. To me, what you say here implies a danger of downtime. Would it be more accurate and/or less scary to say that the modification will cause a deployment (or even "a rolling deployment")?